flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/FailureHandlingResult.java

```diff
@@ -122,7 +135,20 @@ public boolean canRestart() {
 	 * @return result of a set of tasks to restart to recover from the failure
 	 */
 	public static FailureHandlingResult restartable(Set<ExecutionVertexID> verticesToRestart, long restartDelayMS) {
```

Comment:
Minor: If we make `globalFailure` a param instead of adding new factory methods, seems the changes in both `FailureHandlingResult` can `ExecutionFailureHandler` can be much simpler.

Reply:
Happy to do that, it is simpler indeed. Originally, I wanted to preserve the "spirit of the class", with its descriptive factory methods. But I think you are right, the simplicity argument here is stronger.